hibernate-reactive-quickstart/src/main/java/org/acme/hibernate/reactive/FruitMutinyResource.java

```diff
@@ -25,65 +25,67 @@
 public class FruitMutinyResource {
 
     @Inject
-    Uni<Mutiny.Session> mutinySession;
+    Mutiny.Session mutinySession;
 
     @GET
     public Uni<Fruit[]> get() {
-        return mutinySession.flatMap( session -> session.createNamedQuery( "Fruits.findAll", Fruit.class ).getResultList() )
-                .map( fruits -> fruits.toArray( new Fruit[fruits.size()] ) );
+        return mutinySession
+                .createNamedQuery( "Fruits.findAll", Fruit.class ).getResultList()
+                .onItem().apply( fruits -> fruits.toArray( new Fruit[fruits.size()] ) );
     }
 
     @GET
     @Path("{id}")
     public Uni<Fruit> getSingle(@PathParam Integer id) {
-        return mutinySession
-                .flatMap( session -> session.find( Fruit.class, id ) );
+        return mutinySession.find( Fruit.class, id );
     }
 
     @POST
     public Uni<Response> create(Fruit fruit) {
         if (fruit.getId() != null) {
             throw new WebApplicationException("Id was invalidly set on request.", 422);
         }
+
         return mutinySession
-                .flatMap( session -> session.persist( fruit ) )
-                .flatMap( session -> session.flush() )
-                .map( ignore -> Response.ok( fruit ).status( 201 ).build() );
+                .persist( fruit )
+                .onItem().produceUni( session -> mutinySession.flush() )
+                .onItem().apply( ignore -> Response.ok( fruit ).status( 201 ).build() );
     }
 
     @PUT
     @Path("{id}")
-    public Uni<Fruit>  update(@PathParam Integer id, Fruit fruit) {
+    public Uni<Response> update(@PathParam Integer id, Fruit fruit) {
         if (fruit.getName() == null) {
             throw new WebApplicationException("Fruit Name was not set on request.", 422);
         }
 
         return mutinySession
-                .flatMap( session -> session.find( Fruit.class, id )
-                        .flatMap( entity -> {
-                            if (entity == null) {
-                                throw new WebApplicationException("Fruit with id of " + id + " does not exist.", 404);
-                            }
-                            entity.setName( fruit.getName() );
-                            return session.flush()
-                                    .map( ignore -> entity );
-                        } )
-                );
+                .find( Fruit.class, id )
+                // Found
+                .onItem().ifNotNull()
+                    .produceUni( entity -> {
+                        entity.setName( fruit.getName() );
+                        return mutinySession.flush()
+                                .onItem().apply( ignore -> Response.ok( entity ).build() );
+                    } )
+                // Not found
+                .onItem().ifNull()
```

Comment:
I have to admit that even with help from indentation, it's not clear to me to which element of the chain the "ifNull" applies here..

Reply:
What do you think it apply to?

Reply:
> What do you think it apply to?

I just realized that I might sound more confrontational than I intended. I'm just trying to understand what you find confusing so that I can think of a better approach

Reply:
Mine was  just a thought about the `ifNull` API, I'm not sure if there is any thing you can do about it.

In particular, it loolks the "if null" check could apply on the last produced item: the `Response` object

Reply:
Do you think something like this is less confusing?

        // Update the entity
        Function<Fruit, Uni<Response>> update = entity -> {
            entity.setName(fruit.getName());
            return mutinySession.flush()
                    .onItem().apply(ignore -> Response.ok( entity ).build());
        };

        return mutinySession
                .find( Fruit.class, id )
                      // If entity exists then
                    .onItem().ifNotNull()
                        .produceUni(update)
                    // else
                    .onItem().ifNull()
                        .continueWith(Response.ok().status( 404 ).build());

Reply:
I guess it's not correct though, this works because update never returns null so the comment is a bit misleading

Reply:
I checked the documentation and this is the way it is supposed to be used so I think we are good: https://smallrye.io/smallrye-mutiny/#_how_do_i_handle_null

I will extract the function because I think it makes the code more readable.